internal/tiltfile/tiltfile_state.go

```diff
@@ -1479,6 +1483,35 @@ func (s *tiltfileState) translateLocal() ([]model.Manifest, error) {
 	return result, nil
 }
 
+//this deals with alerting the user if there are multiple instances of a
+//the same yaml entity (same name:kind:namespace:group) which may have been
+//defined in the same way or in multiple ways.
+func (s *tiltfileState) warnDuplicateYamlEntities(m model.Manifest) {
+	deployTarget := m.K8sTarget()
+	displayNames := deployTarget.DisplayNames
+
+	duplicates := make(map[string]int)
+	//when the same entity appears more than once, Tilt handles it by storing it as name:kind:namespace:group:n where n is
+	//the nth repeat of that entity, so as to not cause execution to halt. In the future, it might be better to surface an
+	//error about the duplicated resource closer to the actual manifest assembly.
+	for _, name := range displayNames {
+		tempName := strings.Split(name, ":")
+		if len(tempName) > 4 { //this means there's an n which was appended at the manifest assembly because of multiple same entities present
+			duplicates[strings.Join(tempName[:4], ":")]++
+		}
+	}
+
+	//get the names of all the duplicates
+	var duplicateNames []string
+	for key := range duplicates {
```

Comment:
heh there's no reason for `duplicates` to be a map any more, you could cut out this step by just storing duplicate names in a slice from the start :)

Reply:
Ooooh I tried that, but then before I append a particular element to the slice, I'd need to check if its already there right? And since there isn't an easy way to check that like in Python (`if elem in list:`), I thought it'll probably be almost the same (in terms of computational time) to just keep the dictionary aha

Reply:
......you're totally right, I forgot how computers worked 😅  ignore this comment!

Reply:
No worries! I really wish Go had an `in` operator...would make slice stuff a looooot easier haha